lib/galaxy/workflow/scheduling_manager.py

```diff
@@ -326,9 +333,39 @@ def __schedule(self, workflow_scheduler_id, workflow_scheduler):
             if not self.monitor_running:
                 return
 
+    def __attempt_materialize(self, workflow_invocation, session) -> bool:
+        try:
+            hdas_to_materialize = workflow_invocation.inputs_requiring_materialization()
+            for hda in hdas_to_materialize:
+                user = RequestUser(user_id=workflow_invocation.history.user_id)
+                task_request = MaterializeDatasetInstanceTaskRequest(
+                    user=user,
+                    history_id=workflow_invocation.history.id,
+                    source="hda",
+                    content=hda.id,
+                    validate_hashes=True,
+                )
+                self.app.hda_manager.materialize(task_request, in_place=True)
+            # place back into ready and let it proceed normally on next iteration?
+            workflow_invocation.set_state(model.WorkflowInvocation.states.READY)
+            session.add(workflow_invocation)
+            session.commit()
+            return True
+        except Exception as e:
+            log.info(f"Failed to materialize dataset for workflow {workflow_invocation.id} - {e}")
```

Comment:
```suggestion
            log.exception(f"Failed to materialize dataset for workflow {workflow_invocation.id}")
```
I assume this shouldn't ever happen and everything here is entirely unexpected ?

Reply:
Yeah -  I think I wrote that as info before I ran the tests that ensured things were mostly working.